beacon_node/network/src/router/mod.rs

```diff
@@ -256,23 +256,45 @@ impl<T: BeaconChainTypes> Router<T> {
                     }
                 }
             }
-            PubsubMessage::VoluntaryExit(_exit) => {
-                // TODO: Apply more sophisticated validation
-                self.propagate_message(id, peer_id.clone());
-                // TODO: Handle exits
-                debug!(self.log, "Received a voluntary exit"; "peer_id" => format!("{}", peer_id) );
+            PubsubMessage::VoluntaryExit(exit) => {
+                debug!(self.log, "Received a voluntary exit"; "peer_id" => format!("{}", peer_id));
+                if let Some(verified_exit) = self
+                    .processor
+                    .verify_voluntary_exit_for_gossip(&peer_id, *exit)
+                {
+                    self.processor.import_verified_voluntary_exit(verified_exit);
```

Comment:
I don't think it makes any difference at the moment, because imports are disgustingly fast (just a hashmap insert). I guess the question is: if imports become slower in future, should we propagate first? In which case I think the answer is yes, so I'll swap it around.